### ISSUE TITLE
twisted.conch.insults.insults.ServerProtocol.nextLine does not work with gnome-terminal

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#3189]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)| @exarkun reported|
|-|-|
|Trac ID|trac#3189|
|Type|defect|
|Created|2008-04-18 00:51:06Z|
|Branch|https://github.com/twisted/twisted/tree/insults-nextline-3189|

With gnome-terminal 2.18.2 (at least), when the cursor is on the last row of the screen, nextLine doesn't cause everything to scroll up.  This is visible with `twisted/conch/stdio.py`, where the prompts just stack up further and further to the right once the cursor is on the last row.

<details><summary>Searchable metadata</summary>

```
trac-id__3189 3189
type__defect defect
reporter__exarkun exarkun
priority__normal normal
milestone__ 
branch__branches_insults_nextline_3189 branches/insults-nextline-3189
branch_author__ 
status__closed closed
resolution__fixed fixed
component__conch conch
keywords__ 
time__1208479866000000 1208479866000000
changetime__1208868730000000 1208868730000000
version__None None
owner__ 
cc__ralphm
```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_1"></a>|@exarkun commented|
|-|-|

(In [23323]) Branching to 'insults-nextline-3189'

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_2"></a>|@exarkun removed owner|
|-|-|

Fixed, perhaps.

**Comment by twisted-trac** [comment]
|[<img alt="ralphm's avatar" src="https://avatars.githubusercontent.com/u/116741?s=50" width="50" height="50">](https://github.com/ralphm)<a name="note_3"></a>|@ralphm set owner to @exarkun|
|-|-|

For me this indeed fixes the problem I had with gnome-terminal in combination with a manhole SSH session. Tested with gnome-terminal 2.22.1, which is the most recent released version.

Changes look good to me. Please merge.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_5"></a>|@exarkun set status to `closed`|
|-|-|

(In [23377]) Merge insults-nextline-3189

Author: exarkun
Reviewer: ralphm
Fixes: [#3189](https://github.com/twisted/twisted/issues/3189)

Change ServerProtocol.nextLine to use \n instead of ESC E
since the former is supported correctly more widely than
is the latter.  Fix handling of several newline cases in
the in-memory terminal emulator,
twisted.conch.insults.helper.TerminalBuffer.

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
